src/libYARP_dev/src/yarp/dev/Lidar2DDeviceBase.h

```diff
@@ -61,24 +64,30 @@ class YARP_dev_API Lidar2DDeviceBase: public yarp::dev::IRangefinder2D
     //constructor
     Lidar2DDeviceBase();
 
+protected:
+    //utility methods called by laser devices
+    virtual bool updateLidarData();
+    virtual bool updateTimestamp();
+    virtual bool applyLimitsOnLaserData();
+    virtual bool updateLogic() = 0;
+
 public:
     //IRangefinder2D interface
-    bool getRawData(yarp::sig::Vector& data) override;
-    bool getLaserMeasurement(std::vector<LaserMeasurementData>& data) override;
-    bool getDeviceStatus(Device_status& status) override;
-    bool getDeviceInfo(std::string& device_info) override;
-    bool getDistanceRange(double& min, double& max) override;
-    bool getScanLimits(double& min, double& max) override;
-    bool getHorizontalResolution(double& step) override;
-    bool getScanRate(double& rate) override;
+    virtual bool getRawData(yarp::sig::Vector& data) override;
+    virtual bool getLaserMeasurement(std::vector<LaserMeasurementData>& data) override;
+    virtual bool getDeviceStatus(Device_status& status) override;
+    virtual bool getDeviceInfo(std::string& device_info) override;
+    virtual bool getDistanceRange(double& min, double& max) override;
+    virtual bool getScanLimits(double& min, double& max) override;
+    virtual bool getHorizontalResolution(double& step) override;
+    virtual bool getScanRate(double& rate) override;
```

Comment:
Virtual is automatically implied by override, as reported by the [`modernize-use-override` Clang-tidy check](https://clang.llvm.org/extra/clang-tidy/checks/modernize-use-override.html).
I suggest to leave this as it is.

Reply:
done